api/app/Http/Controllers/Api/UserController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Models\User;
+use App\Services\Weather\Contracts\WeatherProvider;
+use App\Services\Weather\WeatherResponse;
+use App\Services\Weather\WeatherException;
+use Illuminate\Http\JsonResponse;
+
+class UserController extends Controller
+{
+    public function __construct(private WeatherProvider $weather) {}
+
+    /**
+     * GET /api/users
+     * Returns all users with their current weather (if available).
+     */
+    public function index(): JsonResponse
+    {
+        $users = User::query()->latest()->take(20)->get();
+
+        $data = $users->map(fn (User $u) => $this->serializeUser($u))->all();
```

Comment:
A cool little Laravel trick you could have used to make this more readable is to move the `serializeUser` method to the `User` model and then call it as a [collection Higher Order Message](https://laravel.com/docs/12.x/collections#higher-order-messages):

```suggestion
        $data = $users->map->serializeUser()->all();
```

Reply:
That looks cleaner.